src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -222,7 +238,6 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
 
             // Configure DI using Startup type
             bool usingCustomServiceProvider;
```

Comment:
Maybe not...this PR is messy on purpose to keep changes small for this initial "do you like this at all?" review :)